feathr-example/installment-payments_credit_card_balance.ipynb

```diff
@@ -0,0 +1,1073 @@
+    "        df_['AMT_INSTALMENT'] = pd.to_numeric(df_['AMT_INSTALMENT'])\n",
+    "        df_['AMT_PAYMENT'] = pd.to_numeric(df_['AMT_PAYMENT'])\n",
+    "        \n",
+    "        df_['INSTALMENT_MISSED'] = (df_['AMT_INSTALMENT'] > df_['AMT_PAYMENT']).astype(int)\n",
+    "        df_['AMT_UNPAID'] = df_['AMT_INSTALMENT'] - df_['AMT_PAYMENT']\n",
+    "        df_['PERC_UNPAID'] = df_['AMT_UNPAID']/df_['AMT_INSTALMENT']\n",
+    "        \n",
+    "        df_ = df_.fillna(0)\n",
+    "        agg = df_.groupby(\"SK_ID_CURR\")\n",
+    "        # percentage of missed payments\n",
+    "        missed_instalments = agg['INSTALMENT_MISSED'].agg(lambda x: x.sum()/x.count()). \\\n",
+    "            reset_index().set_index(\"SK_ID_CURR\")\n",
+    "        # percentage of payments difference for each missed payment\n",
+    "        avg_percent_unpaid = agg['PERC_UNPAID'].mean().reset_index().set_index(\"SK_ID_CURR\")\n",
+    "        # average payments difference for each missed payment\n",
+    "        avg_unpaid = agg['AMT_UNPAID'].mean().reset_index().set_index(\"SK_ID_CURR\")\n",
+    "        final_df = missed_instalments\n",
+    "        final_df = final_df.join(avg_percent_unpaid, on='SK_ID_CURR')\n",
+    "        final_df = final_df.join(avg_unpaid,on=\"SK_ID_CURR\")\n",
+    "        return final_df\n",
+    "\n",
+    "    # add a TRAN_DATE column with a static date\n",
+    "    df = df.withColumn(\"TRAN_DATE\", lit(datetime.datetime(2021,1,1,11,34,44).strftime('%Y-%m-%d %X')))\n",
+    "\n",
+    "    df_org = df.toPandas()\n",
+    "\n",
+    "    df_aggAvgInstalments = aggAvgInstalments(df_org)\n",
+    "\n",
+    "    # results df would be merge to the original df\n",
+    "    df_result = pd.merge(df_org, df_aggAvgInstalments, on=\"SK_ID_CURR\", how=\"left\")\n",
+    "    # merging df with same column name would result a columnname with a suffix of `_x` and `_y`.\n",
+    "    # Renaming the column name with suffix `_x` to retain the original column name\n",
+    "    df_result.columns = df_result.columns.str.rstrip(\"_x\")\n",
+    "\n",
+    "    \n",
+    "    # convert panda to spark dataframe\n",
+    "    spark_session = sql.SparkSession.builder.appName(\"pdf to sdf\").getOrCreate()\n",
+    "    \n",
+    "    return spark_session.createDataFrame(df_result)  \n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Feature definition for Installments payments"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "\n",
+    "# source for pass through features\n",
+    "# \"TRAN_DATE\" column created on on the \"datasource_prepocessing\" method.\n",
+    "installments_payments_source_core = HdfsSource(name=\"installmentsPaymentsSourceCore\",\n",
+    "                          path=\"abfss://feathrhomecreditcafs@feathrhomecreditcasto.dfs.core.windows.net/home_credit_data/installments_payments.csv\",\n",
```

Comment:
Azure account reference.

Reply:
Moved the resource prefix to a variable.